dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -352,6 +379,17 @@ await using (enumerator.ConfigureAwait(false))
         }
     }
 
+    /// <summary>
+    /// Creates a new <see cref="KernelFunction"/> object that is a copy of the current instance
+    /// but the <see cref="KernelFunctionMetadata"/> has the plugin name set.
+    /// </summary>
+    /// <param name="pluginName">The name of the plugin this function instance will be added to.</param>
+    /// <remarks>
+    /// This method should only be used to create a new instance of a <see cref="KernelFunction"/> when adding
+    /// a function to a <see cref="KernelPlugin"/>.
+    /// </remarks>
+    public abstract KernelFunction CreateForPlugin(string pluginName);
```

Comment:
This is more of a Clone rather than Create method, right? Maybe `CloneWithPluginName` or something along those lines?

Reply:
That was my original name and no-one liked it but I will revert this.

Reply:
In this case, how about just `Clone`?
In the future it will be easier to add similar overload that will accept something different than single string (if there will be such requirement).

For example, instead of this:
```csharp
CloneWithPluginName(string pluginName);
CloneWithParameters(string pluginName, IReadOnlyList<KernelParameterMetadata> parameters);
```

It will be like this:
```csharp
Clone(string pluginName);
Clone(string pluginName, IReadOnlyList<KernelParameterMetadata> parameters);
```

Feel free to skip if you think it doesn't make sense :)